brainiak/fcma/preprocessing.py

```diff
@@ -83,8 +92,72 @@ def _separate_epochs(activity_data, epoch_list):
     return raw_data, labels
 
 
+def _randomize_single_subject(data, seed=None):
+    """Randomly permute the voxels of the subject.
+
+     The subject is organized as Voxel x TR,
+     this method shuffles the voxel dimension.
+
+    Parameters
+    ----------
+    data: 2D array in shape [nVoxels, nTRs]
+        Activity data.
+    seed: Optional[int]
+        Seed for random state used implicitly for shuffling.
+
+    Returns
+    data: 2D array in shape [nVoxels, nTRs]
+        Activity data with the voxel dimension shuffled.
+    """
+    if seed is not None:
+        np.random.seed(seed)
+    np.random.shuffle(data)
+    return data
+
+
+def _randomize_subject_list(data_list, random):
+    """Randomly permute the voxels of a subject list.
+
+     The method shuffles the subject one by one according to
+     the random type. If RandomType.NORANDOM, return the
+     original list.
+
+    Parameters
+    ----------
+    data_list: list of 2D array in shape [nVxels, nTRs]
+        Activity data list.
+    random: RandomType
+        Randomization type.
+
+    Returns
+    data_list: list of 2D array in shape [nVxels, nTRs]
+        (Randomized) activity data list.
+    """
+    if random == RandomType.REPRODUCIBLE:
+        for i in range(len(data_list)):
+            data_list[i] = _randomize_single_subject(data_list[i], seed=i)
+    elif random == RandomType.UNREPRODUCIBLE:
+        for i in range(len(data_list)):
+            data_list[i] = _randomize_single_subject(data_list[i])
```

Comment:
This is better, but still mixed usage. If you're going to do the change in place, don't return a value. If you're going to return a value, it should be a new value. So if you want to use in-place, it should look like:
```
for subject in data_list:
    _randomize_single_subject(subject)
```
So neither _randomize_single_subject nor this function should return a value. Also, note since you're modifying in place, you don't need to assign to indexes, so you can use the simpler for loop.